homeworks/MarlQ/1-halfedge/README.md

```diff
@@ -0,0 +1,60 @@
+# Homework 1: Implement the Halfedge Structure
+
+**Task 1. Build Halfedge-based mesh structure in the [`buildMesh`](./src/geometry/halfedge.ts) method, then render the mesh using the provided [software rasterizer](./src/).**
+
+![](./assets/expect.png)
+
+- _Hint: Don't work directly on a complex mesh, but try to start with a single triangle._
+- Think about how to turn the learned data structure to code.
+- Read blender's [developer documentation](https://wiki.blender.org/wiki/Source/Modeling/BMesh/Design), try to answer these questions:
+- What is the blender's internal mesh data structure?
+- How it works and what are the differences compare to Halfedge-based mesh strcture?
+
+Write the answers here:
+
+```
+Blender's internal mesh data structure is BMesh.
+In contrast to Halfedge-based mesh structures, BMesh is designed to represent non-manifold meshes.
+BMesh's entities consist of Vertices, Edges, Loops, and Faces.
+Similar to Halfedge, BMesh uses double-linked, circular lists, but unlike Halfedge, the stored Cycles BMesh uses can have more complicated structures, like the Disk Cycle, which is independent from face data, making it able to represent wire edges and non-manifold faces.
+
+```
+
+**Task 2. Write a reproducer of teaser image in [teaser.py](./teaser.py).**
+
+- Tutorial: [The official Blender 2.8 Tutorial](https://www.youtube.com/playlist?list=PLa1F2ddGya_-UvuAqHAksYnB0qL9yWDO6), [Blender Guru's Beginner Tutorial Series](https://www.youtube.com/playlist?list=PLjEaoINr3zgEq0u2MzVgAaHEBt--xLB6U), etc.
+- Checkout Blender's [Python API documentation](https://docs.blender.org/api/current/index.html).
+- Use [the bunny model](./assets/bunny.obj) from GitHub repository, write a Python script that reproduce the course [teaser figure](../assets/teaser.png). To execute the script:
+
+    ```sh
+    $ blender -b -P ./teaser.py
+    ```
+
+**Task 3. Think about your individual project ideas.**
+
+- _Hint: See submitted projects from last year in https://github.com/mimuc/gp/tree/ws2021#final-projects_
+
+Write initial ideas here:
+
+```
+- Volume-preserving squash (and stretch) without physics simulation or animation, based on object collisions.
+- neural network processing (e.g. for smoothing)
```

Comment:
As discussed in the precense, these two ideas are awesome but may require a large amount of work. Please balance the time investment when going on these ideas.